streams/src/test/java/org/apache/kafka/streams/integration/ResetPartitionTimeIntegrationTest.java

```diff
@@ -0,0 +1,279 @@
+        streamsConfig.put(StreamsConfig.POLL_MS_CONFIG, Integer.toString(DEFAULT_TIMEOUT));
+        streamsConfig.put(StreamsConfig.COMMIT_INTERVAL_MS_CONFIG, Integer.toString(DEFAULT_TIMEOUT));
+        streamsConfig.put(StreamsConfig.PROCESSING_GUARANTEE_CONFIG, eosEnabled ? EXACTLY_ONCE : AT_LEAST_ONCE);
+        streamsConfig.put(StreamsConfig.STATE_DIR_CONFIG, TestUtils.tempDirectory().getPath());
+        
+
+        KafkaStreams driver = getStartedStreams(streamsConfig, builder, true);
+        try {
+            // start sending some records to have partition time committed 
+            produceSynchronouslyToPartitionZero(
+                input,
+                asList(
+                    new KeyValueTimestamp<>("k3", "v3", 5000)
+                )
+            );
+            verifyOutput(
+                outputRaw,
+                asList(
+                    new KeyValueTimestamp<>("k3", 1L, 5000)
+                )
+            );
+            assertThat(lastRecordedTimestamp, is(5000L));
+            lastRecordedTimestamp = -1L;
+            Thread.sleep(1000); // wait for commit to finish
+
+            // restart && reset the driver
+            driver.close();
+            assertThat(driver.state(), is(KafkaStreams.State.NOT_RUNNING));
+            driver.cleanUp();
+            driver = new KafkaStreams(builder.build(), streamsConfig);
+            driver.start();
+
+            // resend some records and retrieve the last committed timestamp
+            produceSynchronouslyToPartitionZero(
+                input,
+                asList(
+                    new KeyValueTimestamp<>("k4", "v4", 4998),
+                    new KeyValueTimestamp<>("k5", "v5", 4999)
+                )
+            );
+            verifyOutput(
+                outputRaw,
+                asList(
+                    new KeyValueTimestamp<>("k4", 1L, 4998),
+                    new KeyValueTimestamp<>("k5", 1L, 4999)
+                )
+            );
+            // verify that the lastRecordedTimestamp is 5000
+            assertThat(lastRecordedTimestamp, is(5000L));
+
+            metadataValidator.raiseExceptionIfAny();
+
+        } finally {
+            driver.close();
+            cleanStateAfterTest(CLUSTER, driver);
+        }
+    }
+
+    public static final class MaxTimestampExtractor implements TimestampExtractor {
+        @Override
+        public long extract(final ConsumerRecord<Object, Object> record, final long maxTimestamp) {
+            if (record.timestamp() > maxTimestamp) {
+                // we do so because maxTimestamp will by RecordQueue logic turn into this record's timestamp
+                lastRecordedTimestamp = record.timestamp();
+            } else {
+                lastRecordedTimestamp = maxTimestamp;
+            }
+            System.out.println("TimestampExtractor: partition time being returned: " + record.timestamp() + " and lastRecordedTimestamp: " + lastRecordedTimestamp + " for partition " + 
```

Comment:
Seem not to be required?

Reply:
Yeah, forgot to remove it during previous debugging.